doc/user/content/guides/python.md

```diff
@@ -0,0 +1,196 @@
+
+Materialize is **PostgreSQL-compatible**, which means that Python applications can use any existing PostgreSQL client to interact with Materialize as if it were a PostgreSQL database. In this guide, we'll use [psycopg2 PostgreSQL database adapter](https://pypi.org/project/psycopg2/) to connect to Materialize and issue PostgreSQL commands.
+
+## Connect
+
+You connect to Materialize the same way you [connect to PostgreSQL with `psycopg2`](https://www.psycopg.org/docs/usage.html).
+
+### Local Instance
+
+You can connect to a local Materialize instance just as you would connect to a PostgreSQL instance:
+
+```python
+#!/usr/bin/env python3
+
+import psycopg2
+import sys
+
+dsn = "postgresql://materialize@localhost:6875/materialize?sslmode=disable"
+conn = psycopg2.connect(dsn)
+```
+
+### Materialize Cloud Instance
+
+Download your instance's certificate files from the Materialize Cloud [Connect](/cloud/connect-to-cloud/) dialog and specify the path to each file in the [connection string parameters](https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNSTRING). Replace `MY_INSTANCE_ID` in the connection string property with your Materialize Cloud instance ID.
+
+```python
+#!/usr/bin/env python3
+
+import psycopg2
+import sys
+
+dsn = "postgresql://materialize@MY_INSTANCE_ID:6875/materialize?sslmode=verify-full&sslcert=materialize.crt&sslkey=materialize.key&sslrootcert=ca.crt"
+conn = psycopg2.connect(dsn)
+```
+
+## Stream
+
+To take full advantage of incrementally updated materialized views from a Python application, instead of [querying](#query) Materialize for the state of a view at a point in time, use [a `TAIL` statement](/sql/tail/) to request a stream of updates as the view changes.
+
+To read a stream of updates from an existing materialized view, open a long-lived transaction with `BEGIN` and use [`TAIL` with `FETCH`](/sql/tail/#tailing-with-fetch) to repeatedly fetch all changes to the view since the last query.
+
+```python
+#!/usr/bin/env python3
+
+import psycopg2
+import sys
+
+dsn = "postgresql://materialize@localhost:6875/materialize?sslmode=disable"
+conn = psycopg2.connect(dsn)
+
+with conn.cursor() as cur:
+    cur.execute("DECLARE c CURSOR FOR TAIL my_view")
+    while True:
+        cur.execute("FETCH ALL c")
+        for row in cur:
+            print(row)
+```
+
+The [TAIL Output format](/sql/tail/#output) of `res.rows` is an array of view update objects. When a row of a tailed view is **updated,** two objects will show up in the `rows` array:
+
+```python
+    ...
+(Decimal('1648737001490'), 1, 'my_value_1')
+(Decimal('1648737001490'), 1, 'my_value_2')
+(Decimal('1648737001490'), 1, 'my_value_3')
+(Decimal('1648737065479'), -1, 'my_value_3')
+(Decimal('1648737065479'), 1, 'my_value_4')
+    ...
+```
+
+A `mz_diff` value of `-1` indicates Materialize is deleting one row with the included values.  An update is just a deletion (`mz_diff: '-1'`) and an insertion (`mz_diff: '1'`) with the same `mz_timestamp`.
+
```

Comment:
Add the "Streaming with psycopg3" section from https://materialize.com/docs/sql/tail/#streaming-with-python-and-psycopg3

````suggestion

### Streaming with psycopg3

{{< warning >}}
psycopg3 is not yet stable.
The example here could break if their API changes.
{{< /warning >}}

Although psycopg3 can function identically as the psycopg2 example above,
it also has a `stream` feature where rows are not buffered and we can thus use `TAIL` directly.

```python
#!/usr/bin/env python3

import psycopg3
import sys

dsn = "postgresql://materialize@localhost:6875/materialize?sslmode=disable"
conn = psycopg3.connect(dsn)

conn = psycopg3.connect(dsn)
with conn.cursor() as cur:
    for row in cur.stream("TAIL t"):
        print(row)
```
````